docs/ingressroute.md

```diff
@@ -0,0 +1,613 @@
+
+#### Load Balancing Strategy
+
+Each upstream service can have a load balancing strategy applied to determine which of its Endpoints is selected for the request. The following list are the options available to choose from:
+
+- RoundRobin: Each healthy upstream Endpoint is selected in round robin order (Default strategy if none selected)
+- WeightedLeastRequest: The least request strategy uses an O(1) algorithm which selects two random healthy Endpoints and picks the Endpoint which has fewer active requests. Note: This algorithm is simple and sufficient for load testing. It should not be used where true weighted least request behavior is desired
+- Ring hash: The ring/modulo hash load balancer implements consistent hashing to upstream Endpoints
+- Maglev: The Maglev strategy implements consistent hashing to upstream Endpoints
+- Random: The random strategy selects a random healthy Endpoints
+
+More information on the load balancing strategy can be found in [Envoy's documentation](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/load_balancing.html).
+
+The following example IngressRoute defines the strategy for Service `s2-strategy` as `WeightedLeastRequest`.  Service `s1-strategy` does not have an explicit strategy defined so it will use the strategy of `RoundRobin`.
+
+```yaml
+# lb-strategy.ingressroute.yaml
+apiVersion: contour.heptio.com/v1beta1
+kind: IngressRoute
+metadata: 
+  name: lb-strategy
+  namespace: default
+spec: 
+  virtualhost:
+    fqdn: strategy.bar.com
+  routes: 
+    - match: /
+      services: 
+        - name: s1-strategy
+          port: 80
+        - name: s2-strategy
+          port: 80
+          strategy: WeightedLeastRequest
+```
+
+#### IngressRoute Default Load Balancing Strategy (Not supported in beta.1)
+
+In order to reduce the amount of duplicated configuration, the IngressRoute specification supports a default Strategy that will be applied to all Services.  You may still override this default on a per-Service basis.
+
+In this example, Services `s1-def-strategy` and `s2-def-strategy` will both have requests distributed across their Endpoints using the WeightedLeastRequest strategy.  Service `s3-def-strategy` will have requests distributed randomly.
+
+```yaml
+# default-lb-strategy.ingressroute.yaml
+apiVersion: contour.heptio.com/v1beta1
+kind: IngressRoute
+metadata: 
+  name: default-lb-strategy
+  namespace: default
+spec: 
+  virtualhost:
+    fqdn: default-strategy.bar.com
+  strategy: WeightedLeastRequest # Default LB algorithm to be applied to services
+  routes: 
+    - match: /
+      services: 
+        - name: s1-def-strategy
+          port: 80
+        - name: s2-def-strategy
+          port: 80
+        - name: s3-def-strategy
+          port: 80
+          strategy: Random # Overrides default LB algorithm for only this Service
+```
+
+#### Active Health Checking
```

Comment:
Suggestion: maybe rephrase this as "Per upstream health checking", then we can avoid talking about Kubernete's pod health check and just talk about the healthcheck: key in the ingressroute doc.